Tests fail with Python 3.6: AssertionError: Expected '_log' to be called once. Called 0 times.

This is really weird. The following test fails when run via python3.6 on my system but works fine with 3.5 and 3.7:

```
$ python3.6 --version
Python 3.6.9
$ tox -e py36
py36 installed: IMAPClient==2.2.0a0
py36 run-test-pre: PYTHONHASHSEED='1616548834'
py36 run-test: commands[0] | python setup.py test
/tmp/imapclient/.tox/py36/lib/python3.6/site-packages/setuptools/dist.py:388: UserWarning: Normalizing '2.2.0-alpha' to '2.2.0a0'
  normalized_version,
running test
Searching for six
Best match: six 1.12.0
Processing six-1.12.0-py3.6.egg

Using /tmp/imapclient/.eggs/six-1.12.0-py3.6.egg
running egg_info
writing IMAPClient.egg-info/PKG-INFO
writing dependency_links to IMAPClient.egg-info/dependency_links.txt
writing requirements to IMAPClient.egg-info/requires.txt
writing top-level names to IMAPClient.egg-info/top_level.txt
reading manifest file 'IMAPClient.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no files found matching 'HACKING.rst'
warning: no files found matching 'AUTHORS'
warning: no files found matching 'THANKS'
warning: no files found matching 'ez_setup.py'
warning: no files found matching '*' under directory 'imapclient/examples'
writing manifest file 'IMAPClient.egg-info/SOURCES.txt'
running build_ext
test_alpha (tests.test_version.TestVersionString) ... ok
test_beta_point (tests.test_version.TestVersionString) ... ok
test_dot_oh (tests.test_version.TestVersionString) ... ok
test_minor (tests.test_version.TestVersionString) ... ok
test_point_release (tests.test_version.TestVersionString) ... ok
test_assert_imap_protocol (tests.test_util_functions.TestAssertIMAPProtocol) ... ok
test_assert_imap_protocol_with_message (tests.test_util_functions.TestAssertIMAPProtocol) ... ok
test_binary (tests.test_util_functions.Test_join_message_ids) ... ok
test_binary_non_numeric (tests.test_util_functions.Test_join_message_ids) ... ok
test_int (tests.test_util_functions.Test_join_message_ids) ... ok
test_iter (tests.test_util_functions.Test_join_message_ids) ... ok
test_mixed_list (tests.test_util_functions.Test_join_message_ids) ... ok
test_tuple (tests.test_util_functions.Test_join_message_ids) ... ok
test_unicode (tests.test_util_functions.Test_join_message_ids) ... ok
test_unicode_non_numeric (tests.test_util_functions.Test_join_message_ids) ... ok
test_None (tests.test_util_functions.Test_normalise_search_criteria) ... ok
test_binary (tests.test_util_functions.Test_normalise_search_criteria) ... ok
test_binary_with_charset (tests.test_util_functions.Test_normalise_search_criteria) ... ok
test_empty (tests.test_util_functions.Test_normalise_search_criteria) ... ok
test_ints (tests.test_util_functions.Test_normalise_search_criteria) ... ok
test_list (tests.test_util_functions.Test_normalise_search_criteria) ... ok
test_mixed_list (tests.test_util_functions.Test_normalise_search_criteria) ... ok
test_no_quoting_when_criteria_given_as_string (tests.test_util_functions.Test_normalise_search_criteria) ... ok
test_quoting (tests.test_util_functions.Test_normalise_search_criteria) ... ok
test_tuple (tests.test_util_functions.Test_normalise_search_criteria) ... ok
test_unicode (tests.test_util_functions.Test_normalise_search_criteria) ... ok
test_unicode_with_charset (tests.test_util_functions.Test_normalise_search_criteria) ... ok
test_binary (tests.test_util_functions.Test_normalise_text_list) ... ok
test_iter (tests.test_util_functions.Test_normalise_text_list) ... ok
test_list (tests.test_util_functions.Test_normalise_text_list) ... ok
test_mixed_list (tests.test_util_functions.Test_normalise_text_list) ... ok
test_tuple (tests.test_util_functions.Test_normalise_text_list) ... ok
test_unicode (tests.test_util_functions.Test_normalise_text_list) ... ok
test_binary (tests.test_util_functions.Test_seq_to_parenstr) ... ok
test_iter (tests.test_util_functions.Test_seq_to_parenstr) ... ok
test_list (tests.test_util_functions.Test_seq_to_parenstr) ... ok
test_mixed_list (tests.test_util_functions.Test_seq_to_parenstr) ... ok
test_tuple (tests.test_util_functions.Test_seq_to_parenstr) ... ok
test_unicode (tests.test_util_functions.Test_seq_to_parenstr) ... ok
test_binary (tests.test_util_functions.Test_seq_to_parenstr_upper) ... ok
test_iter (tests.test_util_functions.Test_seq_to_parenstr_upper) ... ok
test_list (tests.test_util_functions.Test_seq_to_parenstr_upper) ... ok
test_mixed_list (tests.test_util_functions.Test_seq_to_parenstr_upper) ... ok
test_tuple (tests.test_util_functions.Test_seq_to_parenstr_upper) ... ok
test_unicode (tests.test_util_functions.Test_seq_to_parenstr_upper) ... ok
test_all_args (tests.test_thread.TestThread) ... ok
test_defaults (tests.test_thread.TestThread) ... ok
test_no_thread_support (tests.test_thread.TestThread) ... ok
test_unsupported_algorithm (tests.test_thread.TestThread) ... ok
test_add (tests.test_store.TestFlags) ... ok
test_get (tests.test_store.TestFlags) ... ok
test_remove (tests.test_store.TestFlags) ... ok
test_set (tests.test_store.TestFlags) ... ok
test_flags_are_bytes (tests.test_store.TestFlagsConsts) ... ok
test_add (tests.test_store.TestGmailLabels) ... ok
test_get (tests.test_store.TestGmailLabels) ... ok
test_remove (tests.test_store.TestGmailLabels) ... ok
test_set (tests.test_store.TestGmailLabels) ... ok
test_command_fails (tests.test_starttls.TestStarttls) ... ok
test_fails_if_called_twice (tests.test_starttls.TestStarttls) ... ok
test_fails_if_ssl_true (tests.test_starttls.TestStarttls) ... ok
test_works (tests.test_starttls.TestStarttls) ... ok
test_all_args (tests.test_sort.TestSort) ... ok
test_multiple_criteria (tests.test_sort.TestSort) ... ok
test_no_support (tests.test_sort.TestSort) ... ok
test_single_criteria (tests.test_sort.TestSort) ... ok
test_bytes_query (tests.test_search.TestGmailSearch) ... ok
test_bytes_query_with_charset (tests.test_search.TestGmailSearch) ... ok
test_unicode_criteria_with_charset (tests.test_search.TestGmailSearch) ... ok
test_bytes_criteria (tests.test_search.TestSearch) ... ok
test_bytes_criteria_with_charset (tests.test_search.TestSearch) ... ok
test_modseq (tests.test_search.TestSearch) ... ok
test_nested (tests.test_search.TestSearch) ... ok
test_nested_empty (tests.test_search.TestSearch) ... ok
test_nested_multiple (tests.test_search.TestSearch) ... ok
test_nested_tuple (tests.test_search.TestSearch) ... ok
test_no_results (tests.test_search.TestSearch) ... ok
test_quoting (tests.test_search.TestSearch) ... ok
test_search_custom_exception_with_invalid_list (tests.test_search.TestSearch) ... ok
test_search_custom_exception_with_invalid_text (tests.test_search.TestSearch) ... ok
test_single (tests.test_search.TestSearch) ... ok
test_unicode_criteria (tests.test_search.TestSearch) ... ok
test_unicode_criteria_with_charset (tests.test_search.TestSearch) ... ok
test_with_date (tests.test_search.TestSearch) ... ok
test_with_datetime (tests.test_search.TestSearch) ... ok
test_zero_length_quoting (tests.test_search.TestSearch) ... ok
test_Address_str (tests.test_response_parser.TestParseFetchResponse) ... ok
test_Address_str_ignores_encoding_error (tests.test_response_parser.TestParseFetchResponse) ... ok
test_BODY (tests.test_response_parser.TestParseFetchResponse) ... ok
test_BODYSTRUCTURE (tests.test_response_parser.TestParseFetchResponse) ... ok
test_BODY_HEADER_FIELDS (tests.test_response_parser.TestParseFetchResponse) ... ok
test_ENVELOPE (tests.test_response_parser.TestParseFetchResponse) ... ok
test_ENVELOPE_with_empty_addresses (tests.test_response_parser.TestParseFetchResponse) ... ok
test_ENVELOPE_with_invalid_date (tests.test_response_parser.TestParseFetchResponse) ... ok
test_ENVELOPE_with_no_date (tests.test_response_parser.TestParseFetchResponse) ... ok
test_FLAGS (tests.test_response_parser.TestParseFetchResponse) ... ok
test_INTERNALDATE (tests.test_response_parser.TestParseFetchResponse) ... ok
test_INTERNALDATE_NIL (tests.test_response_parser.TestParseFetchResponse) ... ok
test_INTERNALDATE_normalised (tests.test_response_parser.TestParseFetchResponse) ... ok
test_UID (tests.test_response_parser.TestParseFetchResponse) ... ok
test_bad_UID (tests.test_response_parser.TestParseFetchResponse) ... ok
test_bad_data (tests.test_response_parser.TestParseFetchResponse) ... ok
test_bad_msgid (tests.test_response_parser.TestParseFetchResponse) ... ok
test_basic (tests.test_response_parser.TestParseFetchResponse) ... ok
test_literals (tests.test_response_parser.TestParseFetchResponse) ... ok
test_literals_and_keys_with_square_brackets (tests.test_response_parser.TestParseFetchResponse) ... ok
test_missing_data (tests.test_response_parser.TestParseFetchResponse) ... ok
test_mixed_types (tests.test_response_parser.TestParseFetchResponse) ... ok
test_multiple_messages (tests.test_response_parser.TestParseFetchResponse) ... ok
test_none_special_case (tests.test_response_parser.TestParseFetchResponse) ... ok
test_not_uid_is_key (tests.test_response_parser.TestParseFetchResponse) ... ok
test_odd_pairs (tests.test_response_parser.TestParseFetchResponse) ... ok
test_partial_fetch (tests.test_response_parser.TestParseFetchResponse) ... ok
test_same_message_appearing_multiple_times (tests.test_response_parser.TestParseFetchResponse) ... ok
test_simple_pairs (tests.test_response_parser.TestParseFetchResponse) ... ok
test_basic (tests.test_response_parser.TestParseMessageList) ... ok
test_modseq (tests.test_response_parser.TestParseMessageList) ... ok
test_modseq_interleaved (tests.test_response_parser.TestParseMessageList) ... ok
test_modseq_no_space (tests.test_response_parser.TestParseMessageList) ... ok
test_one_id (tests.test_response_parser.TestParseMessageList) ... ok
test_bad_literal (tests.test_response_parser.TestParseResponse) ... ok
test_bad_quoting (tests.test_response_parser.TestParseResponse) ... ok
test_complex_mixed (tests.test_response_parser.TestParseResponse) ... ok
test_deeper_nest_tuple (tests.test_response_parser.TestParseResponse) ... ok
test_empty_tuple (tests.test_response_parser.TestParseResponse) ... ok
test_envelopey (tests.test_response_parser.TestParseResponse) ... ok
test_envelopey_quoted (tests.test_response_parser.TestParseResponse) ... ok
test_incomplete_tuple (tests.test_response_parser.TestParseResponse) ... ok
test_int (tests.test_response_parser.TestParseResponse) ... ok
test_int_and_tuple (tests.test_response_parser.TestParseResponse) ... ok
test_literal (tests.test_response_parser.TestParseResponse) ... ok
test_literal_with_more (tests.test_response_parser.TestParseResponse) ... ok
test_nested_tuple (tests.test_response_parser.TestParseResponse) ... ok
test_nil (tests.test_response_parser.TestParseResponse) ... ok
test_quoted_specials (tests.test_response_parser.TestParseResponse) ... ok
test_square_brackets (tests.test_response_parser.TestParseResponse) ... ok
test_string (tests.test_response_parser.TestParseResponse) ... ok
test_tuple (tests.test_response_parser.TestParseResponse) ... ok
test_unquoted (tests.test_response_parser.TestParseResponse) ... ok
test_escaping (tests.test_response_lexer.TestTokenSource) ... ok
test_invalid_escape (tests.test_response_lexer.TestTokenSource) ... ok
test_lists (tests.test_response_lexer.TestTokenSource) ... ok
test_literal (tests.test_response_lexer.TestTokenSource) ... ok
test_literals (tests.test_response_lexer.TestTokenSource) ... ok
test_multiple_inputs (tests.test_response_lexer.TestTokenSource) ... ok
test_no_escaping_in_square_brackets (tests.test_response_lexer.TestTokenSource) ... ok
test_one_token (tests.test_response_lexer.TestTokenSource) ... ok
test_quoted_strings (tests.test_response_lexer.TestTokenSource) ... ok
test_simple_tokens (tests.test_response_lexer.TestTokenSource) ... ok
test_square_brackets (tests.test_response_lexer.TestTokenSource) ... ok
test_unmatched_square_brackets (tests.test_response_lexer.TestTokenSource) ... ok
test_unterminated_strings (tests.test_response_lexer.TestTokenSource) ... ok
test_whitespace (tests.test_response_lexer.TestTokenSource) ... ok
test_SSL (tests.test_init.TestInit) ... ok
test_plain (tests.test_init.TestInit) ... ok
test_ssl_and_stream_is_error (tests.test_init.TestInit) ... ok
test_stream (tests.test_init.TestInit) ... ok
test_stream_and_port_is_error (tests.test_init.TestInit) ... ok
test_getacl (tests.test_imapclient.TestAclMethods) ... ok
test_setacl (tests.test_imapclient.TestAclMethods) ... ok
test_with_msg_time (tests.test_imapclient.TestAppend) ... ok
test_without_msg_time (tests.test_imapclient.TestAppend) ... ok
test_caching (tests.test_imapclient.TestCapabilities) ... ok
test_decorator (tests.test_imapclient.TestCapabilities) ... ok
test_has_capability (tests.test_imapclient.TestCapabilities) ... ok
test_post_auth_request (tests.test_imapclient.TestCapabilities) ... ok
test_preauth (tests.test_imapclient.TestCapabilities) ... ok
test_server_returned_capability_after_auth (tests.test_imapclient.TestCapabilities) ... ok
test_with_starttls (tests.test_imapclient.TestCapabilities) ... ok
test_context_manager (tests.test_imapclient.TestContextManager) ... ok
test_context_manager_fail_closing (tests.test_imapclient.TestContextManager) ... ok
test_exception_inside_context_manager (tests.test_imapclient.TestContextManager) ... ok
test_IMAP_is_patched (tests.test_imapclient.TestDebugLogging) ... ok
test_redacted_password (tests.test_imapclient.TestDebugLogging) ... FAIL
test_expunge (tests.test_imapclient.TestExpunge) ... ok
test_id_expunge (tests.test_imapclient.TestExpunge) ... ok
test_find_special_folder_with_special_use (tests.test_imapclient.TestFindSpecialFolder) ... ok
test_find_special_folder_with_special_use_single_flag (tests.test_imapclient.TestFindSpecialFolder) ... ok
test_find_special_folder_without_special_use_nor_namespace (tests.test_imapclient.TestFindSpecialFolder) ... ok
test_id (tests.test_imapclient.TestId) ... ok
test_invalid_parameters (tests.test_imapclient.TestId) ... ok
test_no_support (tests.test_imapclient.TestId) ... ok
test_consume_until_tagged_response (tests.test_imapclient.TestIdleAndNoop) ... ok
test_idle (tests.test_imapclient.TestIdleAndNoop) ... ok
test_idle_check_blocking (tests.test_imapclient.TestIdleAndNoop) ... ok
test_idle_check_timeout (tests.test_imapclient.TestIdleAndNoop) ... ok
test_idle_check_with_data (tests.test_imapclient.TestIdleAndNoop) ... ok
test_idle_done (tests.test_imapclient.TestIdleAndNoop) ... ok
test_noop (tests.test_imapclient.TestIdleAndNoop) ... ok
test_blanks (tests.test_imapclient.TestListFolders) ... ok
test_empty_response (tests.test_imapclient.TestListFolders) ... ok
test_folder_encode_off (tests.test_imapclient.TestListFolders) ... ok
test_funky_characters (tests.test_imapclient.TestListFolders) ... ok
test_list_folders (tests.test_imapclient.TestListFolders) ... ok
test_list_folders_NO (tests.test_imapclient.TestListFolders) ... ok
test_list_sub_folders (tests.test_imapclient.TestListFolders) ... ok
test_list_sub_folders_NO (tests.test_imapclient.TestListFolders) ... ok
test_mixed (tests.test_imapclient.TestListFolders) ... ok
test_quoted_specials (tests.test_imapclient.TestListFolders) ... ok
test_simple (tests.test_imapclient.TestListFolders) ... ok
test_unquoted_numeric_folder_name (tests.test_imapclient.TestListFolders) ... ok
test_unquoted_numeric_folder_name_parsed_as_long (tests.test_imapclient.TestListFolders) ... ok
test_utf7_decoding (tests.test_imapclient.TestListFolders) ... ok
test_without_quotes (tests.test_imapclient.TestListFolders) ... ok
test_complex (tests.test_imapclient.TestNamespace) ... ok
test_folder_decoding (tests.test_imapclient.TestNamespace) ... ok
test_other_only (tests.test_imapclient.TestNamespace) ... ok
test_simple (tests.test_imapclient.TestNamespace) ... ok
test_without_folder_decoding (tests.test_imapclient.TestNamespace) ... ok
test_tagged_response_with_parse_error (tests.test_imapclient.TestProtocolError) ... ok
test__get_quota (tests.test_imapclient.TestQuota) ... ok
test_get_quota_root (tests.test_imapclient.TestQuota) ... ok
test_parse_quota (tests.test_imapclient.TestQuota) ... ok
test_set_quota (tests.test_imapclient.TestQuota) ... ok
test_complex (tests.test_imapclient.TestRawCommand) ... ok
test_embedded_literal (tests.test_imapclient.TestRawCommand) ... ok
test_failed_continuation_wait (tests.test_imapclient.TestRawCommand) ... ok
test_invalid_input_type (tests.test_imapclient.TestRawCommand) ... ok
test_literal_at_end (tests.test_imapclient.TestRawCommand) ... ok
test_multiple_literals (tests.test_imapclient.TestRawCommand) ... ok
test_not_uid (tests.test_imapclient.TestRawCommand) ... ok
test_plain (tests.test_imapclient.TestRawCommand) ... ok
test_normal (tests.test_imapclient.TestSelectFolder) ... ok
test_unselect (tests.test_imapclient.TestSelectFolder) ... ok
test_shutdown (tests.test_imapclient.TestShutdown) ... ok
test_default (tests.test_imapclient.TestTimeNormalisation) ... ok
test_pass_through (tests.test_imapclient.TestTimeNormalisation) ... ok
test_decode (tests.test_imap_utf7.IMAP4UTF7TestCase) ... ok
test_encode (tests.test_imap_utf7.IMAP4UTF7TestCase) ... ok
test_printable_singletons (tests.test_imap_utf7.IMAP4UTF7TestCase) ... ok
test_basic (tests.test_folder_status.TestFolderStatus) ... ok
test_extra_response (tests.test_folder_status.TestFolderStatus) ... ok
test_literal (tests.test_folder_status.TestFolderStatus) ... ok
test_GMT (tests.test_fixed_offset.TestFixedOffset) ... ok
test_for_system_DST_active (tests.test_fixed_offset.TestFixedOffset) ... ok
test_for_system_DST_not_active (tests.test_fixed_offset.TestFixedOffset) ... ok
test_for_system_no_DST (tests.test_fixed_offset.TestFixedOffset) ... ok
test_negative (tests.test_fixed_offset.TestFixedOffset) ... ok
test_positive (tests.test_fixed_offset.TestFixedOffset) ... ok
test_failed1 (tests.test_enable.TestEnable) ... ok
test_failed2 (tests.test_enable.TestEnable) ... ok
test_multiple (tests.test_enable.TestEnable) ... ok
test_success (tests.test_enable.TestEnable) ... ok
test_wrong_state (tests.test_enable.TestEnable) ... ok
test_basic (tests.test_datetime_util.TestCriteriaDateFormatting) ... ok
test_single_digit_day (tests.test_datetime_util.TestCriteriaDateFormatting) ... ok
test_with_timezone (tests.test_datetime_util.TestDatetimeToINTERNALDATE) ... ok
test_without_timezone (tests.test_datetime_util.TestDatetimeToINTERNALDATE) ... ok
test_dots_for_time_separator (tests.test_datetime_util.TestParsing) ... ok
test_internaldate_style (tests.test_datetime_util.TestParsing) ... ok
test_invalid (tests.test_datetime_util.TestParsing) ... ok
test_rfc822_style (tests.test_datetime_util.TestParsing) ... ok
test_fail (tests.test_auth.TestPlainLogin) ... ok
test_simple (tests.test_auth.TestPlainLogin) ... ok
test_with_authorization_identity (tests.test_auth.TestPlainLogin) ... ok

======================================================================
FAIL: test_redacted_password (tests.test_imapclient.TestDebugLogging)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/imapclient/tests/test_imapclient.py", line 620, in test_redacted_password
    extra={}
  File "/usr/lib64/python3.6/unittest/mock.py", line 824, in assert_called_once_with
    raise AssertionError(msg)
AssertionError: Expected '_log' to be called once. Called 0 times.

----------------------------------------------------------------------
Ran 255 tests in 0.327s

FAILED (failures=1)
Test failed: <unittest.runner.TextTestResult run=255 errors=0 failures=1>
error: Test failed: <unittest.runner.TextTestResult run=255 errors=0 failures=1>
ERROR: InvocationError for command /tmp/imapclient/.tox/py36/bin/python setup.py test (exited with code 1)
_____________________________________________________________________ summary _____________________________________________________________________
ERROR:   py36: commands failed
```

--- a/tests/test_imapclient.py
+++ b/tests/test_imapclient.py
@@ -604,8 +604,8 @@ class TestDebugLogging(IMAPClientTest):
         adapter = IMAPlibLoggerAdapter(logger_mock, dict())
         if six.PY3:
             adapter.info("""> b'ICHH1 LOGIN foo@bar.org "secret"'""")
-            if sys.version_info >= (3, 7):
-                # LoggerAdapter in Python 3.7+ calls logger.log()
+            if sys.version_info >= (3, 6, 4):
+                # LoggerAdapter in Python 3.6.4+ calls logger.log()
                 logger_mock.log.assert_called_once_with(
                     logging.INFO,
                     "> b'ICHH1 LOGIN **REDACTED**",

